aspnetcore/fundamentals/http-logging/samples/8.x/Program.cs

```diff
@@ -59,5 +59,50 @@
 app.MapGet("/", () => "Hello World!");
 
 app.Run();
-#endregion
+// </snippet2>
+#elif THIRD
+// <snippet3>
+using HttpLoggingSample;
+using Microsoft.AspNetCore.HttpLogging;
+
+// <snippet4>
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddHttpLogging(logging =>
+{
+    logging.LoggingFields = HttpLoggingFields.Duration;
+});
+builder.Services.AddHttpLoggingInterceptor<SampleHttpLoggingInterceptor>();
+// </snippet4>
+var app = builder.Build();
+
+if (!app.Environment.IsDevelopment())
+{
+    app.UseExceptionHandler("/Error");
+}
+
+app.UseStaticFiles();
+
+app.UseHttpLogging(); 
+
+app.Use(async (context, next) =>
+{
+    context.Response.Headers["MyResponseHeader"] =
+        new string[] { "My Response Header Value" };
+
+    await next();
+});
+
+app.MapGet("/", () => "Hello World!");
+
+// <snippet5>
+app.MapGet("/duration", [HttpLogging(loggingFields: HttpLoggingFields.Duration)]
+    () => "Hello World! (logging duration)");
+
+app.MapGet("/request", [HttpLogging(loggingFields: HttpLoggingFields.RequestPropertiesAndHeaders)]
```

Comment:
Oh yeah, we also have `.WithHttpLogging(HttpLoggingFields.RequestPropertiesAndHeaders);` for this minimal scenario. The attribute is more for controllers, but it works here too.
https://github.com/dotnet/aspnetcore/blob/82ff9246592477abf4ff57ac905a368cbd99ff20/src/Middleware/HttpLogging/src/HttpLoggingEndpointConventionBuilderExtensions.cs#L23